accepted/0000-subscription-matching-in-public-clouds.md

```diff
@@ -0,0 +1,287 @@
+- Feature Name: Subscription matching in public clouds
+- Start Date: 2019-08-14
+- RFC PR: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This RFC set the guides for supporting "Subscription Matching" on virtual instances created on Public Clouds like AWS, Azure or Google Compute Engine. Exploring and gatehering virtual instances from those Public Clouds enables "Subscription Matching" to properly match subscriptions for "1-2 Virtual Machines". In the context of Public Clouds, "Unlimited VMs" subscriptions are not allowed by the "[Terms & Conditions](https://www.suse.com/products/terms_and_conditions.pdf)".
+
+# Motivation
+[motivation]: #motivation
+
+Currently the subscription matcher does not work for VMs that are running in public clouds. Our `virtual-host-gatherer` tool doesn't support exploring those public clouds, so there is no effective way to provide the needed information that makes subscription matcher be aware of the virtualization environments and therefore match the subscription and benefits from "1-2 Virtual Machine" licencing.
+
+This RFC defines the procedures to enabling "Subscriptions Matching" on different Public Clouds providers.
+
+Also provides implementation details for supporting the following modules for `virtual-host-gatherer`:
+
+- A generic (JSON-file based) module that enables Administrators to provide its own custom virtual instances definition.
+- A module for gathering virtual instances from Azure Cloud (based on libcloud).
+- A module for gathering virtual instances from AWS (based on libcloud).
+- A module for gathering virtual instances from Google Compute Engine (based on libcloud).
+
```

Comment:
Alibaba?

Reply:
I added "Alibaba Cloud" to the list other potential providers to support (since metadata API is available): https://github.com/uyuni-project/uyuni-rfc/blame/master-subscription-matching-in-public-clouds/accepted/0000-subscription-matching-in-public-clouds.md#L168

I understood we're not implementing support on "Alibaba Cloud" in the first iteration of the implementation, that's why I didn't add a particular example for it. But, as discussed, implementing it using the "libcloud" approach should be trivial.